ansible/fhir-stress-test.yml

```diff
@@ -19,87 +24,112 @@
-#            cidr_ip: 0.0.0.0/0
-#    
-#    - name: EC2 - Provision 'bluebutton-stress' client
-#      ec2:
-#        key_name: "{{ ec2_key_name }}"
-#        group:
-#          - default
-#          - ssh-all
-#        instance_type: "{{ ec2_instance_type }}"
-#        image: "{{ ami_id_ubuntu }}"
-#        region: "{{ region }}"
-#        wait: true
-#        exact_count: 1
-#        count_tag:
-#          Name: bluebutton-stress-client
-#          Application: "BlueButton"
-#        instance_tags:
-#          Name: bluebutton-stress-client
-#          Application: "BlueButton"
-#          CreatedBy: "{{ whoami.stdout }}"
-#      register: ec2_client
-#    
-#    - name: EC2 - Provision 'bluebutton-stress' servers
-#      ec2:
-#        key_name: "{{ ec2_key_name }}"
-#        group:
-#          - default
-#          - ssh-all
-#        instance_type: "{{ ec2_instance_type }}"
-#        image: "{{ ami_id_ubuntu }}"
-#        region: "{{ region }}"
-#        wait: true
-#        # The default of 300s here wasn't cutting it for 8 servers.
-#        wait_timeout: 600
-#        exact_count: "{{ num_servers }}"
-#        count_tag:
-#          Name: bluebutton-stress-server
-#          Application: "BlueButton"
-#        instance_tags:
-#          Name: bluebutton-stress-server
-#          Application: "BlueButton"
-#          CreatedBy: "{{ whoami.stdout }}"
-#      register: ec2_servers
-#    
-#    - name: Create inventory group 'client'
-#      add_host: 
-#        hostname: "{{ item.public_ip }}"
-#        groupname: client
-#      with_items: ec2_client.instances
-#
-#    - name: Create inventory group 'servers'
-#      add_host:
-#        hostname: "{{ item.public_ip }}"
-#        groupname: servers
-#      with_items: ec2_servers.instances
-#
-#    - name: Wait for SSH to come up
-#      wait_for:
-#        host: "{{ item.public_dns_name }}"
-#        port: 22
-#        timeout: 320
-#        state: started
-#      with_flattened:
-#        - ec2_client.instances
-#        - ec2_servers.instances
+    - name: Create Local Support Directory 
+      file:
+        path: support
+        state: directory
+
+    # The EC2 instances created within this AWS environment are not internet
+    # facing. So, assuming the host running this script is internet facing,
+    # download the jmeter package locally so it can be uploaded to the instances
+    # from this host.  Another strategy would be to just create an AMI used by
+    # the EC2 instances that already includes this package.
+    - name: Download JMeter locally
+      get_url:
+        url: "http://archive.apache.org/dist/jmeter/binaries/apache-jmeter-{{ jmeter_version }}.tgz"
+        dest: "support/apache-jmeter-{{ jmeter_version }}.tgz"
+
+    - name: Acquire a session token
+      sts_session_token:
+        profile: "{{ profile }}"
+        region: "{{ region }}"
+        mfa_serial_number: "{{ mfa_sn }}"
+        mfa_token: "{{ mfa }}"
+      register: session_credentials
```

Comment:
This totally works for now, but keep in mind that if/when we run all this from Jenkins, it won't be needed from there. Jenkins will instead leverage the permissions assigned to its EC2 instance profile.

Reply:
Understood, but I needed to test it! 👍